### PR TITLE
Overhaul task polling functions

### PR DESCRIPTION
**Pull request by Ichimonji10**
Do the following:

* Let the `poll_spawned_tasks` function use a `pulp_smash.api.Client`
  object when polling Pulp's API, rather than directly using the
  underlying Requests library. This makes polling code more concise, and
  it makes some other simplifications possible.
* Don't set a default value for the `pulp_host` parameter. If a user
  passes no value for this parameter, just pass the `None` through to
  `pulp_smash.api.Client`, which will provide a default value if needed.
  In short: this change makes the logic for providing a default value
  for `pulp_host` more DRY.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Tested against Pulp 2.16.1 nightly with:

```bash
for package in pulp_smash.tests.pulp2.{platform,python}.api_v2; do
    python -m unittest discover -fv "$package"
done
```

Pulp 3 testing is in progress.

**Comment by kersommoura** [comment]
Tested against Pulp3 installed from source.

``` python
 python -m unittest discover -f pulp_smash.tests.pulp3
```
